File: BTT4/BTT4 (BTT3 UPDATE)/BTT3/menu.cpp
#include "menu.h"
#include "SeenMessGUI.h"

using namespace std;

void menu::Displaymenu()
{
	LogInGUI LogInGUIMenu1;
	RegisterGUI RegisterGUIMenu1;

	cout << "Hello and Welcom to my project" << endl;
	cout << "if you have account, plesase input 1 and input 2 if you want to register account :)) if input differ 1 or 2 program will be out " << endl;
	//	system("PAUSE");

	string choose;
	cin >> choose;
	if (choose == "1") {
		string userName;
		string passWord;
		std::cout << "moi ban nhap userName: \n";
		std::cin >> userName;
		std::cout << "moi ban nhap passWord: \n";
		std::cin >> passWord;
		if (LogInGUIMenu1.InputUserNameandPassword(userName, passWord) != "NOTHING") {

			process(LogInGUIMenu1.InputUserNameandPassword(userName, passWord));
		}
		else {
			cout << "no have account!!!, please input again!!" << endl;
			Displaymenu();
		}

		//	process(LogInGUIMian.InputUserNameandPassword());
	}if (choose == "2") {
		RegisterGUIMenu1.InputPersonalInfo();
		//	cout << "please input again!!" << endl;
		Displaymenu();
	}
	if (choose != "1" && choose != "2") {

		exit(0);

	}
}



void menu::process(string userID)
{

	LogOutGUI LogOutGUIMenu1;
	RequestDislayMesswithFriendGUI RequestDislayMesswithFriendGUIMenu1;
	DisplayMessUserSentGUI DisplayDetailMessGUIMenu1;
	SentMessGUI SentMessGUIMenu1;
	AddFriendGUI AddFriendGUIMenu1;
	DisplayFriendListGUI DisplayFriendListGUIMenu1;
	BlockFriendGUI BlockFriendGUIMenu1;
	SeenMessGUI SeenMessGUIMenu1;
	int b=0;
	int n;
//	char key = getch();
//	int value = key;
	do {
	
/*		{
		case 2: {
			n = b;
			switch (n)
			{
			case 0: {
				exit(0);
			}
			case 1: {
				RequestDislayMesswithFriendGUIMenu1.RequestDislayMesswithFriend(userID);
				break;
			}
			case 2: {
				DisplayDetailMessGUIMenu1.RequestDisplay(userID);
				break;
			}
			case 3: {
				SentMessGUIMenu1.RequestSentMess(userID);
				break;
			}
			case 4: {
				AddFriendGUIMenu1.InputFriendUserName(userID);
				break;
			}
			case 5: {
				DisplayFriendListGUIMenu1.RequestFriendList(userID);
				break;
			}
			case 6: {
				BlockFriendGUIMenu1.RequestBlockFriendGUI(userID);
				break;
			}
			case 7: {
				SeenMessGUIMenu1.RequestViewMess(userID);
				break;
			}
			case 8: {
				LogOutGUIMenu1.RequestLogOut(userID);
				break;
			}
			}


		}
		case 14: {*/
			cout << "your ID: " << userID << endl;
			cout << "please, choose: " << endl;
			cout << "press ctr+B to back view" << endl;
			cout << "press ctr+N to continue" << endl;
			cout << "0: exit" << endl;
			cout << "1: RequestDislayMesswithFriendGUI" << endl;
			cout << "2: DisplayDetailMessGUI" << endl;
			cout << "3: SentMessGUI" << endl;
			cout << "4: AddFriendGUI" << endl;
			cout << "5: DisplayFriendListGUI" << endl;
			cout << "6: BlockFriendGUI" << endl;
			cout << "7: see seen?" << endl;
			cout << "8: LogOut" << endl;
			//  char key = getch();
			//	int value = key;
			//	if (value == 14 || value != 2) {
			cin >> n;
			b = n;

			switch (n)
			{
			case 0: {
				exit(0);
			}
			case 1: {
				RequestDislayMesswithFriendGUIMenu1.RequestDislayMesswithFriend(userID);
				break;
			}
			case 2: {
				DisplayDetailMessGUIMenu1.RequestDisplay(userID);
				break;
			}
			case 3: {
				SentMessGUIMenu1.RequestSentMess(userID);
				break;
			}
			case 4: {
				AddFriendGUIMenu1.InputFriendUserName(userID);
				break;
			}
			case 5: {
				DisplayFriendListGUIMenu1.RequestFriendList(userID);
				break;
			}
			case 6: {
				BlockFriendGUIMenu1.RequestBlockFriendGUI(userID);
				break;
			}
			case 7: {
				SeenMessGUIMenu1.RequestViewMess(userID);
				break;
			}
			case 8: {
				LogOutGUIMenu1.RequestLogOut(userID);
				break;
			}
			}
		//}
	//	}
	} while (n>0|| n<8);

}

void menu::Choose(string userID)
{
	int b;
	char key = getch();
	int value = key;
	do {
		switch (value)
		{


		case 2: {

		}

		case 14: {

		}
		}
	} while (value != 2 || value != 14);



}





menu::menu()
{
}


menu::~menu()
{
}

File: BTT4_PY1/BTT4_PY1/Menu.py
from RegisterGUI import RegisterGUI
from LogInGUI import LogInGUI
from Controller import Controller
from LogOutGUI import LogOutGUI
from Model import Model
from BlockFriendGUI import BlockFriendGUI
from DisplayFriendListGUI import DisplayFriendListGUI
from DisplayMessUserSentGUI import DisplayMessUserSentGUI
from DisplayDetailMessGUI import DisplayDetailMessGUI
from SentMessGUI import SentMessGUI
from SeenMess import SeenMess
from ManageFriendsInformationGUI import ManageFriendsInformationGUI
from AddFriendGUI import AddFriendGUI

class Menu(object):
	def Process(self,userID):
		sentMessGUI=SentMessGUI
		logOutGUI = LogOutGUI
		blockFriendGUI=BlockFriendGUI
		displayFriendListGUI=DisplayFriendListGUI
		displayMessUserSentGUI =DisplayMessUserSentGUI 
		displayDetailMessGUI =DisplayDetailMessGUI
		seenMess=SeenMess
		manageFriendsInformationGUI=ManageFriendsInformationGUI
		addFriendGUI =AddFriendGUI 
		print("meo")
		print("your ID: ",userID )
		print("please, choose: ")
		#print("press ctr+B to back view")
		#print("press ctr+N to continue")
		print("0: exit")
		print("1: RequestDislayMesswithFriendGUI")
		print("2: DisplayDetailMessGUI")
		print("3: SentMessGUI")
		print("4: AddFriendGUI")
		print("5: DisplayFriendListGUI")
		print("6: BlockFriendGUI")
		print("7: see seen?")
		print("8: LogOut")
		choose=input()
		if(choose=="0"):
			exit()
		if(choose=="1"):
			displayMessUserSentGUI.RequestDisplayMessUserSentGUI(userID)
		if(choose=="2"):
			displayDetailMessGUI.RequestDisplayDetailMessGUI(userID)
		if(choose=="3"):
			sentMessGUI.RequestSentMess(userID)
		if(choose=="4"):
			addFriendGUI.RequestToAddFriend1(userID)
		if(choose=="5"):
			displayFriendListGUI.RequestToDisplayFriendList(userID)
		if(choose=="6"):
			blockFriendGUI.RequestToBlockFriend(userID)
		if(choose=="7"):
			seenMess.RequestViewMess(userID)
		if(choose=="8"):
			logOutGUI.RequestLogOut(userID)
		return self.Process(userID)
	
	def first(self):
		registerGUI= RegisterGUI
		logInGUI=LogInGUI
		print("press 1 to login, 2 to register")
		choose= input("press:" )
		if(choose=="2"):
			registerGUI.RequestRegister()
			return self.first()
		if(choose=="1"):
		
			userID=logInGUI.RequestLogIn()
			if(userID==None):
				print("no acc, press again!!")
				return self.first()
			if(userID!=None):
				self.Process(userID)
				


